bitcoin/build.rs

```diff
@@ -1,4 +1,29 @@
-const MSRV_MINOR: u64 = 63;
+use std::fs;
+
+fn parse_msrv_minor() -> Option<u64> {
+    let cargo_toml =
+        fs::read_to_string("Cargo.toml").expect("Should have been able to read the file");
+
+    let mut msrv_minor = None;
+    for line in cargo_toml.lines() {
+        if line.starts_with("rust-version") {
+            let mut v_line = line.split("=");
+
+            let v = match (v_line.next(), v_line.next()) {
+                (Some(_lhs), Some(rhs)) => rhs,
+                _ => panic!("Unable to parse MSRV"),
+            };
+
+            let mut semvar = v.split(".");
+            msrv_minor = match (semvar.next(), semvar.next()) {
+                (Some(_), Some(m)) => Some(m.parse::<u64>().expect("invalid Rust minor version")),
+                _ => panic!("Unable to parse semvar"),
+            };
+        }
+    }
+
+    msrv_minor
+}
 
```

Comment:
Concept ACK, perhaps we can write it a bit more simply:

```rust
pub fn parse_msrv_minor() -> u64 {
    let cargo_toml = fs::read_to_string("Cargo.toml").expect("manifest exists");

    for line in cargo_toml.lines() {
        if line.starts_with("rust-version") {
            let mut bits = line.split("=");
            bits.next();        // Skip "rust-version" string.
            let semver = bits.next().unwrap();

            let mut nums = semver.split(".");
            nums.next();        // Skip major version number.
            let minor = nums.next().unwrap();
            return minor.parse::<u64>().unwrap();
        }
    }

    panic!("unable to read rust-version field from manifest")
}
```

Reply:
tbh I don't much like doing early returns and the panic! at the end, although the expect message here is def better.

Reply:
Added fixup commit [6eaf2dc](https://github.com/rust-bitcoin/rust-bitcoin/pull/3145/commits/6eaf2dc78c65660d8646c05cc5b70a0aece05a1c)

Reply:
can be auto-squashed if this looks good